web/src/flow/stages/captcha/CaptchaStage.ts

```diff
@@ -45,6 +46,11 @@ export class CaptchaStage extends BaseStage<CaptchaChallenge, CaptchaChallengeRe
     @state()
     scriptElement?: HTMLScriptElement;
 
+    @property()
+    onTokenChange: TokenHandler = (_token: string) => {
+        throw new Error("Client failed to supply a handling function to CaptchaStage");
```

Comment:
Shouldn't the default for this be 
```
this.host?.submit({
    token: token,
});
```

Reply:
Ken and I don't think so, see some discussion above.

(By the rules of refactoring, you're right. I'm altering component contract here and I changed every callsite to match.)

Reply:
I'm mainly asking because when the stage is rendered as default, there's no default handler there, which doesn't seem correct? Or did I miss it

Reply:
This is the point I realize I'm probably missing something. What does "rendered as default" mean here?

My thinking is: this component will always need to be passed `challenge` and `host`. Whoever's supplying that (e.g. the `FlowExecutor`) should also supply `onTokenChange`.

Reply:
Ah yeah I must've overlooked https://github.com/goauthentik/authentik/pull/11711/files#diff-4ed6c41b430eb15195ace2424e99a073db6c9e588f4627cf0f899ef9fad0f084R311-R312, I'd prefer that from the aspect of the flow executor all stages have the same signature so imo it should be the default callback handler on the stage and then it can be overridden when rendered embedded